### PR TITLE
sokol-flex, archive-release: merge archive-release bbappend into the recipe

### PR DESCRIPTION
**Pull request by kergoth**
Replace the old unused `layers` and `downloads` archival with the git-based delivery which is actually used and tested. Also do a bit of recipe cleanup to make it easier to maintain. We may want to break this up into separate files eventually, but this will do for now.

### CODE REVIEW

**Comment by kergoth** [comment]
I'll add a bit more detail to the commit messages in a bit.

**Comment by kergoth** [comment]
The commits now have more detail in the messages, so this is ready for review.

File: meta-sokol-flex-support/recipes-core/meta/archive-release.bb [hunk 1]
```diff
@@ -67,7 +67,7 @@ IMAGE_EXTENSION_live = "hddimg iso"
 # This allows us to build in the automated environment for regression,
 # general testing or simply for availability of extra image types for
 # internal use without necessarily packaging them in the installers.
-ARCHIVE_RELEASE_IMAGE_FSTYPES_EXCLUDE ?= "tar.gz tar.bz2"
+ARCHIVE_RELEASE_IMAGE_FSTYPES_EXCLUDE ?= "tar.gz tar.bz2 tar.xz"
```
[the remark this answers is not in the document]
Comment (abelal) [reply meta-sokol-flex-support/recipes-core/meta/archive-release.bb]: Ah so the next commit handles that... this is fine!

**Review by abelal — APPROVED** [review]
:+1:

File: meta-sokol-flex-support/recipes-core/meta/archive-release.bb (hunk 1 above)
[the remark this answers is not in the document]
Comment (kergoth) [reply meta-sokol-flex-support/recipes-core/meta/archive-release.bb]: I'd certainly be open to simplifying it further, it's obvious this recipe has evolved over time and has a rather lot of variables that control its behavior, probably more than we need anymore.

**Pull request merged**